storage/src/mmr/verification.rs

```diff
@@ -36,7 +36,7 @@ pub struct Proof<H: CHasher> {
 /// A trait that allows generic generation of an MMR inclusion proof.
 pub trait Storage<D: Digest> {
     /// Return the number of elements in the MMR.
-    fn size(&self) -> impl Future<Output = Result<u64, Error>>;
+    fn size(&self) -> u64;
```

Comment:
I'm open to keeping this async for now to preserve flexibility for anyone implementing this trait.

That being said, the size should probably be known before this could be called by any impl...

Reply:
yes, mmr size is a pretty critical property of the structure so we always cache it from the start.